platform/darwin/include/MGLStyle.h

```diff
@@ -4,50 +4,115 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
-/** A collection of convenience methods for creating style URLs of default styles provided by Mapbox. These instances of NSURL are cached. */
+/**
+ A version number identifying the latest released version of the suite of default styles provided by Mapbox. This version number may be passed into one of the “StyleURLWithVersion” class methods of MGLStyle.
+ 
+ The value of this constant is current as of the date on which this SDK was published. Consult the <a href="https://www.mapbox.com/api-documentation/#styles">Mapbox Styles API documentation</a> for the most up-to-date style versioning information.
+ 
+ @warning The value of this constant may change in a future release of the SDK. If you use any feature that depends on a specific implementation detail in a default style, you may use the current value of this constant or the underlying style URL, but do not use the constant itself. Such implementation details may change significantly from version to version.
```

Comment:
I was thinking of it being used for sanity-checking more than anything. For example, a developer might want to assert in a unit test that the current version is still 9, as a reminder to upgrade the application to use the newest default style with the runtime styling API. We’d avoid using this constant anywhere else in documentation or guides. It’s sort of like how this framework (and all others) provides a `MapboxVersionNumber` constant just in case.

But you’re right: the way I’ve worded references to this constant in e.g. `-streetsStyleURLWithVersion:` could definitely lead developers to use it without reading this documentation and understanding the ramifications. If we can’t settle on a safer way to treat this constant, I agree that we should make it private.